### ISSUE TITLE
[package] qt/6.6.0: CMAKE_OSX_DEPLOYMENT_TARGET is always 11.0, os.version has no effect

### ISSUE THREAD

**Issue by tamaskenezlego** [conan-io/conan-center-index#20589]
### Description

When conan is building Qt the `os.version` setting has no effect, the actual `CMAKE_OSX_DEPLOYMENT_TARGET` is always `11.0`.
The reason is that the root `CMakeLists.txt` of Qt is intentionally running the script `QtAutoDetect.cmake` before the `project` command, where the conan toolchain file is invoked and would set `CMAKE_OSX_DEPLOYMENT_TARGET` to `os.version`. `QtAutoDetect.cmake` checks `CMAKE_OSX_DEPLOYMENT_TARGET` (which is undefined at that point) and sets it to `11.0`.

### Package and Environment Details

* Package Name/Version: **qt/6.6.0**
* Operating System+version: **MacOS**

### Conan profile

[settings]
os=Macos
os_build=Macos
arch=armv8
arch_build=armv8
compiler=apple-clang
compiler.version=15
compiler.libcxx=libc++
build_type=Release
[options]
[conf]
[build_requires]
[env]

### Steps to reproduce

conan install -r qt/6.6.0 -pr:b=default -pr:h=default

### Logs

conan log is not relevant

**Comment by jwillikers** [comment]
Does Qt set it as a cache variable? Perhaps we could set it first that way.

**Comment by tamaskenezlego** [comment]
@jwillikers , yes, Qt sets it as a cache variable, this is the line (`qtbase/cmake/QtAutoDetect.cmake:335`)

```CMake
set(CMAKE_OSX_DEPLOYMENT_TARGET "${version}" CACHE STRING "${description}")
```

**Comment by jwillikers** [comment]
Thanks! I've reported this as a bug in conan-io/conan#14988. Conan should be setting it as a cache variable in the first place.

**Comment by stevenwdv** [comment]
This is still an issue with Qt 6.8.3 (which sets it to 12). This leads to warnings in our build, such as:

```
ld: warning: object file (/Users/gitlab/.conan2/p/b/opens7056c2fe7a25c/p/lib/libssl.a[3](libssl-lib-d1_lib.o)) was built for newer 'macOS' version (13.3) than being linked (12.0)
```
(we set `os.version=13.3`)
